### PR TITLE
888061: Add repo to centralise SSL code

### PR DESCRIPTION
**Pull request by rorytorneymf**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=888061

### CODE REVIEW

File: caf-ssl-dropwizard/pom.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            <groupId>io.dropwizard</groupId>
+            <artifactId>dropwizard-core</artifactId>
+            <exclusions>
+                <!-- Excluding this due to the following error:
+                     Found duplicate and different classes in [com.sun.activation:jakarta.activation:2.0.1, jakarta.activation:jakarta.activation-api:2.1.0]
+                 -->
+                <exclusion>
+                    <groupId>com.sun.activation</groupId>
+                    <artifactId>jakarta.activation</artifactId>
+                </exclusion>
```
Comment (dermot-hardy) [inline caf-ssl-dropwizard/pom.xml]: What's going on here?

Comment (rorytorneymf) [reply caf-ssl-dropwizard/pom.xml]: The dependency check was failing so I had to exclude one of them:

```
[INFO] Checking test classpath
[WARNING] Found duplicate and different classes in [com.sun.activation:jakarta.activation:2.0.1, jakarta.activation:jakarta.activation-api:2.1.0]:
[WARNING]   jakarta.activation.ActivationDataFlavor
[WARNING]   jakarta.activation.CommandMap
[WARNING]   jakarta.activation.DataHandler
[WARNING]   jakarta.activation.DataHandlerDataSource
[WARNING]   jakarta.activation.DataSourceDataContentHandler
[WARNING]   jakarta.activation.FileDataSource
[WARNING]   jakarta.activation.MailcapCommandMap
[WARNING]   jakarta.activation.MimeTypeParameterList
[WARNING]   jakarta.activation.MimetypesFileTypeMap
[WARNING]   jakarta.activation.ObjectDataContentHandler
[WARNING]   jakarta.activation.SecuritySupport
[WARNING] Found duplicate classes/resources in compile classpath.
[WARNING] Found duplicate and different classes in [com.sun.activation:jakarta.activation:2.0.1, jakarta.activation:jakarta.activation-api:2.1.0]:
[WARNING]   jakarta.activation.ActivationDataFlavor
[WARNING]   jakarta.activation.CommandMap
[WARNING]   jakarta.activation.DataHandler
[WARNING]   jakarta.activation.DataHandlerDataSource
[WARNING]   jakarta.activation.DataSourceDataContentHandler
[WARNING]   jakarta.activation.FileDataSource
[WARNING]   jakarta.activation.MailcapCommandMap
[WARNING]   jakarta.activation.MimeTypeParameterList
[WARNING]   jakarta.activation.MimetypesFileTypeMap
[WARNING]   jakarta.activation.ObjectDataContentHandler
[WARNING]   jakarta.activation.SecuritySupport
[WARNING] Found duplicate classes/resources in runtime classpath.
[WARNING] Found duplicate and different classes in [com.sun.activation:jakarta.activation:2.0.1, jakarta.activation:jakarta.activation-api:2.1.0]:
[WARNING]   jakarta.activation.ActivationDataFlavor
[WARNING]   jakarta.activation.CommandMap
[WARNING]   jakarta.activation.DataHandler
[WARNING]   jakarta.activation.DataHandlerDataSource
[WARNING]   jakarta.activation.DataSourceDataContentHandler
[WARNING]   jakarta.activation.FileDataSource
[WARNING]   jakarta.activation.MailcapCommandMap
[WARNING]   jakarta.activation.MimeTypeParameterList
[WARNING]   jakarta.activation.MimetypesFileTypeMap
[WARNING]   jakarta.activation.ObjectDataContentHandler
[WARNING]   jakarta.activation.SecuritySupport
[WARNING] Found duplicate classes/resources in test classpath.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for caf-ssl 1.0.0-SNAPSHOT:
[INFO]
[INFO] caf-ssl ............................................ SUCCESS [  8.292 s]
[INFO] caf-ssl-dropwizard ................................. FAILURE [ 10.150 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  21.162 s
[INFO] Finished at: 2024-02-26T15:26:23Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.basepom.maven:duplicate-finder-maven-plugin:2.0.1:check (deepcheck-dependencies) on project caf-ssl-dropwizard: Found duplicate classes/resources! -> [Help 1]
```

image

https://stackoverflow.com/a/77243009

Comment (dermot-hardy) [reply caf-ssl-dropwizard/pom.xml]: Add this to the dependency management section of the root pom:

    <dependency>
        <groupId>jakarta.xml.bind</groupId>
        <artifactId>jakarta.xml.bind-api</artifactId>
        <version>4.0.1</version>
    </dependency>

It should allow the exclusion to be removed.

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        <url>https://github.com/CAFapi/caf-ssl</url>
+    </scm>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>jakarta.xml.bind</groupId>
+                <artifactId>jakarta.xml.bind-api</artifactId>
+                <version>4.0.1</version>
+            </dependency>
```
Comment (dermot-hardy) [inline pom.xml]: Drop this down below the other two as we want to keep dependency management in alphabetical order.

File: README.md [hunk 3]
```diff
@@ -1 +1,3 @@
-# caf-ssl
\ No newline at end of file
+# caf-ssl
+
+This project provides SSL utility code.
```
Comment (dermot-hardy) [inline README.md]: We should add a bit more here.  Like how to use the module, what it's for, etc.  But we can do that in a separate PR if you want to get this merged.

File: caf-ssl-dropwizard/pom.xml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>com.github.cafapi.ssl</groupId>
+        <artifactId>caf-ssl</artifactId>
+        <version>1.0.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>caf-ssl-dropwizard</artifactId>
+    <packaging>jar</packaging>
```
Comment (dermot-hardy) [inline caf-ssl-dropwizard/pom.xml]: Minor but I would drop this as it is the default.

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        <version>5.0.0-431</version>
+    </parent>
+    
+    <inceptionYear>2024</inceptionYear>
+
+    <properties>
+        <copyrightYear>2024</copyrightYear>
+        <copyrightNotice>Copyright ${copyrightYear} Open Text.</copyrightNotice>
+    </properties>
+
```
Comment (dermot-hardy) [inline pom.xml:60]: As this is open-source, we will need `<licenses>` and `<developers>` sections in here to be able to publish a release version of this.  I think we also need `<url>` and `<name>`.  [caf-correlation](https://github.com/CAFapi/caf-correlation) may be a decent example of all the things that are required.  It's a lot more than closed-source but most things are covered by caf-common-parent.

https://central.sonatype.org/publish/requirements/

**Review by dermot-hardy — APPROVED** [review]
